distributed/shuffle/tests/test_no_pandas_import.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+import subprocess
+import sys
+
+import_check_code = """
```

Comment:
I tried a couple other options to avoid this kind of nasty "code in a string" pattern:
1. Mucking with `sys.modules` (setting `pandas` to None, removing `dask.dataframe` and `distributed.shuffle`, then re-importing `distributed.shuffle`). This worked, but felt brittle, because you basically had to list all the places `pandas` might be transitively imported from, and remove them from `sys.modules`.
3. Using `multiprocessing` in `spawn` mode to start a fresh interpreter and run this code. Because `spawn` mode works by sending pickled function, the the function must be importable. And un-picking the function means importing `distributed.shuffle.tests.test_no_pandas_import.func_name_whatever`. Therefore, `distributed.shuffle` gets imported before your function can even execute! So though you can assert that pandas hasn't been imported, if it has been imported, you can't tell from where, which makes the test very annoying to debug.

In the end, the "code in a string" was both much easier to read and more robust than anything else I could come up with.

(Also, if distributed had a mindeps test, we might not need this at all. But it's nice to have anyway for local development.)